core/lib/vlog/src/lib.rs

```diff
@@ -301,6 +318,10 @@ impl ObservabilityBuilder {
             None
         };
 
+        if let Some(rust_backtrace) = self.rust_backtrace {
+            std::env::set_var("RUST_BACKTRACE", rust_backtrace);
```

Comment:
Does this work or rather - does this _always_ work?

Quoting [the same doc you've linked](https://doc.rust-lang.org/std/backtrace/#environment-variables):

>Also note that the state of environment variables is cached once the first backtrace is created, so altering RUST_LIB_BACKTRACE or RUST_BACKTRACE at runtime might not actually change how backtraces are captured.

Reply:
hm.. I'll have a look

Reply:
Based on documentation it's good enough. 
We may lose backtrace during the initialization.
But if initialization was successful we will catch the backtrace

Reply:
> The documentation states that the value that is read at the first backtrace is cached, which means that overriding it afterwise doesn't work, no?

My assumption is that it works like this:
* We don't generate any backtraces during initialization => everything is fine, `RUST_BACKTRACE` is set before it was cached
* We generate a backtrace which is a fatal event during initialization as we don't have "try-catch" patterns there => `RUST_BACKTRACE` is not set, but we are panicking anyway

Still, this feels hacky and unnecessary given that we can just ensure to pass the correct `RUST_BACKTRACE` from upstream. We could maybe add a sanity check with a warning that `RUST_BACKTRACE` is not set and hence `json_panic_handler` is not going to work

Reply:
I have the same feeling as @itegulov 

But in the world, where we don't have `RUST_BACKTRACE` env, I see no other possibilities

Reply:
I had a look at the code of backtrace and yes. 
During `Backtrace::capture` they call method `enabled` that caches this value. 
So, everything should be fine if we set it a bit later and don't capture initialization backtraces

Reply:
If the only concern here is that `json_panic_handler` would work incorrectly, then one can always use [force_capture](https://doc.rust-lang.org/std/backtrace/struct.Backtrace.html#method.force_capture) there, no?

Reply:
But should we be able to remove this functionality?

Reply:
What do you mean? Which functionality?
I suggest using `force_capture` in json panic handler, and otherwise do not affect `RUST_BACKTRACE` in any way from the code (e.g. if it's set it's set).

Reply:
So, you suggest to remove possibility to install `RUST_BACKTRACE` am i right?

Reply:
...from the codebase. Basically, do nothing about this env variable in our code.

Reply:
changed to force